agasc/scripts/update_obs_status.py

```diff
@@ -18,26 +18,32 @@
 import pyyaks.logger
 
 from agasc.supplement.magnitudes import star_obs_catalogs
+from agasc.supplement.utils import save_version
 
 logger = logging.getLogger('agasc.supplement')
 
 
-def add_bad_star(bad_star_ids, bad_star_source, suppl_file, dry_run):
+def add_bad_star(bad_star_ids, bad_star_source, suppl_file, dry_run, create=False):
     """
     Update the 'bad' table of the AGASC supplement.
 
     :param bad_star_ids: int or list
     :param bad_star_source: int or list
     :param suppl_file: pathlib.Path
     :param dry_run: bool
+    :param create: bool
+        Create a supplement file if it does not exist
     """
+    suppl_file = Path(suppl_file)
+
     if not bad_star_ids:
         logger.info('Nothing to update')
         return
 
-    if not Path(suppl_file).exists():
+    if not suppl_file.exists():
+        if not create:
```

Comment:
I'm if we have a create flag, the logger message might not be "warning".  But that's a nit.

Reply:
that's right. I didn't think of that. I just left the message that was there (which I had made inconsistent because it said "creating new file" and then raised an exception)